core/src/main/java/org/radargun/stats/IntervalStatistics.java

```diff
@@ -53,6 +57,16 @@ public void merge(Statistics otherStats) {
       endTime = Math.max(otherStats.getEnd(), endTime);
    }
 
+   // TODO I would like to understand why some stages are calling startRequest without calling the begin first
```

Comment:
begin() should be called only when the warm-up/ramp-up phase finishes, just before we enter the steady-state. We switch to steady-state usually just flipping a volatile boolean and we want to record requests that *finished* within the steady-state; that's why the Requests are created even during warm-up.
Alternative would be recording all requests that *started* within steady-state but that requires some ramp-down waiting for all the in-progress requests to finish.